### PR TITLE
Add argument parsing via clap

### PR DESCRIPTION
**Pull request by TDHolmes**
Adds argument parsing via clap to match the C++ library. Also changed the parallel and scan_mode rust APIs to not take bools as they are not boolean.

### CODE REVIEW

File: src/args.rs [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    let parallel_bool: bool = match parallel {
+        0 => false,
+        _ => true,
+    };
+
+    options.set_hardware_mapping(gpio_mapping);
+    options.set_rows(rows);
+    options.set_cols(cols);
+    options.set_chain_length(chain);
+    options.set_parallel(parallel_bool);
```
Comment (TDHolmes) [inline src/args.rs]: I think the parallel API in general is wrong. Perhaps it was a boolean back when this project started, but you can now have up to three now. Probably should fix that now...

File: src/args.rs [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    let _slowdown_gpio = value_t!(parsed_args, "slowdown-gpio", u32).unwrap();
+
+    // flags
+    let show_refresh: bool = parsed_args.is_present("show-refresh");
+    let inverse: bool = parsed_args.is_present("inverse");
+    let no_hardware_pulse: bool = parsed_args.is_present("no-hardware-pulse");
+    let _daemon: bool = parsed_args.is_present("daemon");
+    let _no_drop_privs: bool = parsed_args.is_present("no-drop-privs");
+
+    let scan_mode_bool: bool = match scan_mode {
```
Comment (BroderickCarlin) [inline src/args.rs]: ‘Let scan_mode_bool: bool = scan_mode != 0;’

Could also do this online on line 105 above 🤷‍♀️

**Comment by TDHolmes** [comment]
#14 can enable the rest of the command line options. Will wait for that to land, and then fix this one.

File: src/options.rs [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+        self.limit_refresh_rate_hz = limit_refresh as c_int;
+    }
+
+    pub fn set_pwm_dither_bits(&mut self, pwm_dither_bits: u32) {
+        self.pwm_dither_bits = pwm_dither_bits as c_int;
+    }
+
+    pub fn set_panel_type(&mut self, panel_type: &str) {
+        unsafe {
+            let _ = CString::from_raw(self.panel_type);
```
Comment (BroderickCarlin) [inline src/options.rs:169]: whats going on here?

**Review by BroderickCarlin — APPROVED** [review]

File: src/options.rs (hunk 3 above)
Comment (TDHolmes) [reply src/options.rs:169]: We need to use `CString::into_raw("string literal")` when passing the string into the C API, which makes it so rust won't deallocate the string rust made. For it to be cleaned up, we need to use `from_raw` to consume and clean it up. [from_raw docs](https://doc.rust-lang.org/std/ffi/struct.CString.html#method.from_raw)

File: src/args.rs (hunk 2 above)
Comment (TDHolmes) [reply src/args.rs]: I ended up changing the API to not be `bool` anyways, but good suggestion!

**Pull request merged**